### PR TITLE
Add an array_flat_map function

### PR DESCRIPTION
**Pull request by ChristophWurst**
More functional and more expressive.

This is my fancy code moment of 2021-04-08.

### CODE REVIEW

**Review by miaulalala — APPROVED** [review]
Fancy fancy

File: lib/Db/MessageMapper.php [hunk 1]
```diff
@@ -178,15 +179,10 @@ public function findUidsForIds(Mailbox $mailbox, array $ids) {
...
-		$results = [];
-		foreach ($chunks as $chunk) {
+		return array_flat_map(function(array $chunk) use ($query) {
 			$query->setParameter('ids', $chunk, IQueryBuilder::PARAM_INT_ARRAY);
-			$results[] = $this->findUids($query);
-		}
-
-		return array_merge(...$results);
+			return $this->findUids($query);
+		}, array_chunk($ids, 1000));
```
Comment (nickvergessen) [inline lib/Db/MessageMapper.php:185]: move array_chunk also inside?

Comment (ChristophWurst) [reply lib/Db/MessageMapper.php:185]: inside the new function? Nah, that is too specific for this use case

**Comment by rullzer** [comment]
Look at mr fancy pants over here

**Pull request merged**